src/main/java/com/limechain/grandpa/state/RoundState.java

```diff
@@ -48,13 +64,83 @@ public BigInteger getThreshold() {
     }
 
     private BigInteger getAuthoritiesTotalWeight() {
-        return voters.stream()
-                .map(Authority::getWeight)
-                .reduce(BigInteger.ZERO, BigInteger::add);
+        return authorities.stream().map(Authority::getWeight).reduce(BigInteger.ZERO, BigInteger::add);
     }
 
     public BigInteger derivePrimary() {
-        var votersCount = BigInteger.valueOf(voters.size());
-        return roundNumber.remainder(votersCount);
+        var authoritiesCount = BigInteger.valueOf(authorities.size());
+        return roundNumber.remainder(authoritiesCount);
+    }
+
+    public void saveGrandpaAuthorities() {
+        repository.save(generateAuthorityKey(DBConstants.AUTHORITY_SET, setId), authorities);
+    }
+
+    public Authority[] fetchGrandpaVoters() {
+        return repository.find(generateAuthorityKey(DBConstants.AUTHORITY_SET, setId), new Authority[0]);
+    }
+
+    public void saveAuthoritySetId() {
+        repository.save(DBConstants.SET_ID, setId);
+    }
+
+    public BigInteger fetchAuthoritiesSetId() {
+        return repository.find(DBConstants.SET_ID, BigInteger.ONE);
+    }
+
+    public void saveLatestRound() {
+        repository.save(DBConstants.LATEST_ROUND, roundNumber);
+    }
+
+    public BigInteger fetchLatestRoundNumber() {
```

Comment:
```suggestion
    public BigInteger fetchLatestRound() {
```
So it follows the same naming as the save.